File: 01/part2.py
#!/bin/env python3

input = []

with open("input", "r") as f:
    input = [int(i) for i in f.read().split("\n")[:-1]]

times_increased = 0

for i in range(len(input)):
    if i+4 <= len(input):
        s = sum(input[i:i+4])
        if s - input[i+3] < s - input[i]:
            times_increased += 1

print(times_increased)

File: 02/part2.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func readInput() []string {
	var ret []string

	file, _ := os.Open("input")
	scanner := bufio.NewScanner(file)

	for scanner.Scan() {
		ret = append(ret, scanner.Text())
	}

	return ret
}

func main() {
	input := readInput()

	horizontal_pos := 0
	depth := 0
  aim := 0

	for _, line := range input {
		cmd := strings.Split(line, " ")
		dist, _ := strconv.Atoi(cmd[1])

		switch cmd[0] {
		case "forward":
			horizontal_pos += dist
      depth += aim * dist
			break
		case "up":
			aim -= dist
			break
		case "down":
			aim += dist
			break
		}
	}

	fmt.Printf("Horizontal position: %d\n", horizontal_pos)
	fmt.Printf("Depth:               %d\n", depth)
  fmt.Printf("Product:             %d\n", horizontal_pos * depth)
}
